### PR TITLE
Add support for CUDA version of LightGBM

### PR DESCRIPTION
**Pull request by IncubatorShokuhou**
## Related Issuse or bug
This PR adds support for CUDA version of LightGBM.

#### Describe the changes you've made
At present, PyCaret only supports OpenCL version of LightGBM( as introduced in `PyCaret on GPU` of `README.md`.
Actually, LightGBM also supports a (experimental) CUDA version(https://github.com/microsoft/LightGBM/tree/master/python-package). To use CUDA version, we only need to set `device="cuda"` in `LGBMRegressor` / `LGBMClassifier`.
I made some changes in `containers/models/classification.py`/`containers/models/regression.py` to make LightGBM works properly with `gpu_param=True` or  `gpu_param="force"` , when CUDA version( instead of OpenCL version) of LightGBM is installed.
PS: 
- To build CUDA version: `pip install lightgbm --install-option=--cuda`
- I don't know if information related to LightGBM on GPU should be changed in README/User guide etc. So I only changed the code.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by IncubatorShokuhou** [comment]
Sorry, I accidently pushed another commit (which fixed https://github.com/pycaret/pycaret/issues/1394) in a same PR...

**Comment by Yard1** [comment]
@IncubatorShokuhou we can keep it here

File: pycaret/containers/models/classification.py [hunk 1]
```diff
@@ -26,6 +26,17 @@
 import pycaret.containers.base_container
 import numpy as np
 
+def CompareVersion(version1: str, version2: str) -> int:\
```
[the remark this answers is not in the document]
Comment (Yard1) [reply pycaret/containers/models/classification.py:29]: Can this be also moved to `pycaret.utils`?

File: pycaret/containers/models/regression.py [hunk 2]
```diff
@@ -24,6 +24,17 @@
 import pycaret.containers.base_container
 import numpy as np
 
+def CompareVersion(version1: str, version2: str) -> int:\
```
Comment (Yard1) [inline pycaret/containers/models/regression.py:27]: Move to pycaret.utils and use the same method everywhere

**Review by Yard1 — CHANGES_REQUESTED** [review]
This would need to be documented in `setup()`, but I am not sure it's a good idea. Would it be possible to do the following logic:
if `use_gpu is True` and the `cuda` version can be used (no exception), use that
otherwise, try the `gpu` version

**Comment by IncubatorShokuhou** [comment]
> This would need to be documented in `setup()`, but I am not sure it's a good idea. Would it be possible to do the following logic:
> if `use_gpu is True` and the `cuda` version can be used (no exception), use that
> otherwise, try the `gpu` version

I would like to suggest to use opencl version before cuda version, as cuda version is comparatively experimental according to LightGBM's document.

File: pycaret/containers/models/classification.py [hunk 3]
```diff
@@ -26,6 +26,17 @@
...
+    """
+    Compare between two versions. 
+    If version1 > version2, return 1;
+    if version1 == version2, return 0;
+    if version1 < version2, return -1.
+    """
+    v1, v2 = ([*map(int, v.split('.'))] for v in (version1, version2))
```
[the remark this answers is not in the document]
Comment (IncubatorShokuhou) [reply pycaret/containers/models/classification.py]: Done. The method of comparing version have been changed to `packaging.version`. BTW, do I need to change the version comparision methods in `xgboost` and `catboost` containers as well?

**Comment by IncubatorShokuhou** [comment]
In addition to `pycaret/containers/models/regression.py` and `pycaret/containers/models/classification.py`, I also modified the version comparison method in `pycaret/internal/tabular.py`, which are all scripts includes version comparsion that can I can find.

**Comment by IncubatorShokuhou** [comment]
> I would like to suggest to try opencl version before cuda version, as cuda version is comparatively experimental according to LightGBM's document.

So I think maybe my  logic in scripts are ok and do not need to be changed, isn't it?

**Comment by Yard1** [comment]
@IncubatorShokuhou This looks good. Have you tested it yourself? We don't have CI for GPU.

**Comment by IncubatorShokuhou** [comment]
> @IncubatorShokuhou This looks good. Have you tested it yourself? We don't have CI for GPU.

Yes, I have tested it on a V100 GPU (from aws), and everything works well.

**Review by Yard1 — APPROVED** [review]
Thanks!

**Pull request merged**